File: clients/kotlin/src/main/kotlin/model/PlayerView.kt
package model

import util.StreamUtil

class PlayerView {
    var myId: Int = 0
    lateinit var game: model.Game
    constructor() {}
    constructor(myId: Int, game: model.Game) {
        this.myId = myId
        this.game = game
    }
    companion object {
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): PlayerView {
            val result = PlayerView()
            result.myId = StreamUtil.readInt(stream)
            result.game = model.Game.readFrom(stream)
            return result
        }
    }
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, myId)
        game.writeTo(stream)
    }
}

File: clients/javascript/model/player-view.js
const Game = require('./game').Game;
class PlayerView {
    constructor(myId, game) {
        this.myId = myId;
        this.game = game;
    }
    static async readFrom(stream) {
        let myId;
        myId = await stream.readInt();
        let game;
        game = await Game.readFrom(stream);
        return new PlayerView(myId, game);
    }
    async writeTo(stream) {
        let myId = this.myId;
        await stream.writeInt(myId);
        let game = this.game;
        await game.writeTo(stream);
    }
}
module.exports = { PlayerView: PlayerView }
